src/webgpu/api/operation/rendering/draw.spec.ts

```diff
@@ -1,15 +1,46 @@
 export const description = `
 Tests for the general aspects of draw/drawIndexed/drawIndirect/drawIndexedIndirect.
 
-TODO: plan and implement
-- draws (note bind group state is not tested here):
-    - various zero-sized draws
-    - draws with vertexCount not aligned to primitive topology (line-list or triangle-list) (should not error)
-    - index buffer is {unset, set}
-    - vertex buffers are {unset, set} (some that the pipeline uses, some it doesn't)
-      (note: to test this, the shader in the pipeline doesn't have to actually use inputs)
-    - x= {draw, drawIndexed, drawIndirect, drawIndexedIndirect}
-- ?
+Primitive topology tested in api/operation/render_pipeline/primitive_topology.spec.ts.
+Index format tested in api/operation/command_buffer/render/state_tracking.spec.ts.
+
+TODO:
+* arguments - Test that draw arguments are passed correctly.
+  Test works by drawing triangles to the screen.
+  Horizontally across the screen are triangles with increasing "primitive id".
+  Vertically down the screen are triangles with increasing instance id.
+  Increasing the |first| param should skip some of the beginning triangles on the horizontal axis.
+  Increasing the |first_instance| param should skip of the beginning triangles on the vertical axis.
+  The vertex buffer contains two side-by-side triangles, and base_vertex is used to offset to select the second.
+  The fragment shader writes out to a storage buffer. If the draw is zero-sized, check that no value is written.
```

Comment:
Does the test ALSO check rendered pixels or does it only check this?

Reply:
Checks the center of the triangles - added something for that